pkg/list/list.go

```diff
@@ -56,7 +41,10 @@ func GetPackagesWithStatus(
 			PackageRepoIndexItem: *item.IndexItem,
 		}
 
-		if !onlyInstalled || item.Package != nil {
+		if (!options.OnlyInstalled || item.Package != nil) &&
+			(!options.OnlyOutdated || (item.Package != nil && item.IndexItem != nil &&
+				item.Package.Spec.PackageInfo.Version != item.IndexItem.LatestVersion)) {
```

Comment:
I think this inner condition lacks another `&&`: `item.Package.Spec.PackageInfo.Version != ""`, otherwise the auto-update-enabled packages will always appear in the list of outdated packages. 

The whole `if` is hard to read in my opinion, it might make sense to store parts of it in bool variables or use functions.